### ISSUE TITLE
Api Key error in ClientBuilder::fromConfig

### ISSUE THREAD

**Issue by science695** [elastic/elasticsearch-php#1074]
### Summary of problem or feature request
Too few arguments to function Elasticsearch\ClientBuilder::setApiKey(), 1 passed in
vendor/elasticsearch/elasticsearch/src/Elasticsearch/ClientBuilder.php on line 192
and exactly 2 expected
StackTrace:

> File: vendor/elasticsearch/elasticsearch/src/Elasticsearch/ClientBuilder.php
> line: 192
> function: setApiKey
> class: ElasticSearch\ClientBuilder
> args: [ "MYAPIKEY" ]
> 
> File: src/lib/ElasticSearch.php
> line: 23
> function: fromConfig
> class: ElasticSearch\ClientBuilder
> type: "::"
> args: [ { "apiKey": "MYAPIKEY", ...} ]

ClientBuilder::fromConfig calls setApiKey from the config option 'ApiKey', however, setApiKey has 2 arguments

I can think of several options that would fix this:

- fromConfig accept an array for ApiKey that sends 2 arguments to setApiKey
- setApiKey handle if api key is sent in joined / base64 encoded form instead of split by id and apikey

### Code snippet of problem
```php
$config['apiKey'] = 'MYAPIKEY';
$instance = ClientBuilder::fromConfig($config);
```

https://github.com/elastic/elasticsearch-php/blob/2e56ea9afdc1a96ab6c147ed7b08a5fdf8f8e62a/src/Elasticsearch/ClientBuilder.php#L186-L193

https://github.com/elastic/elasticsearch-php/blob/2e56ea9afdc1a96ab6c147ed7b08a5fdf8f8e62a/src/Elasticsearch/ClientBuilder.php#L350

- CentOS 7
- PHP 7.3
- ES-PHP 7.9

**Comment by science695** [comment]
My suggestions don't properly handle setApiKey and setHosts...as setHosts wants the array as the first argument, while setApiKey has them separated.

Not sure what solution you would prefer, but should handle both scenarios.

**Comment by science695** [comment]
If there are any others dealing with this issue, I put a fix in it by overriding my own ClientBuilder::fromConfig:

```php
	private function fromConfig($config, $quiet = false){
		$builder = new ClientBuilder;
		foreach ($config as $key => $value) {
			$method = "set$key";
			if (method_exists($builder, $method)) {
				// New case for the ApiKey setting **************************
				if($key == "ApiKey" && is_array($value)){
					call_user_func_array( [ $builder, $method ], $value );
				}
				else {
					$builder->$method($value);
				}
				unset($config[$key]);
			}
		}

		if ($quiet === false && count($config) > 0) {
			$unknown = implode(array_keys($config));
			throw new RuntimeException("Unknown parameters provided: $unknown");
		}
		return $builder->build();
	}
```

**Comment by ezimuel** [comment]
@science695 I provied a fix for this issue in #1076. I used the [variadic function](https://www.php.net/manual/en/functions.arguments.php#functions.variable-arg-list) of PHP to support the ability to pass multiple parameters in `set*`.

**Comment by ezimuel** [comment]
I merged this in 7.x since I would like to release the patch for 7.10 release.

**Comment by science695** [comment]
@ezimuel awesome, thanks

**Comment by ezimuel** [comment]
Just released [7.10.0](https://github.com/elastic/elasticsearch-php/releases/tag/v7.10.0) that includes fix #1074.
